lib/jsonapi/serializer.ex

```diff
@@ -98,26 +96,35 @@ defmodule JSONAPI.Serializer do
   defp include_view(valid_includes, key) when is_list(valid_includes) do
     case Keyword.get(valid_includes, key) do
       {view, :include} -> {view, :include}
-      view -> {view, :include}
+      _view -> false
     end
   end
   defp include_view({view, :include}, _key), do: {view, :include}
-  defp include_view(view, _key), do: {view, :include}
+  defp include_view(_view, _key), do: false
 
   def is_data_loaded?(rel_data) do
     assoc_loaded?(rel_data) && (is_map(rel_data) || (is_list(rel_data) && !Enum.empty?(rel_data)))
   end
 
-  def encode_relation(rel_view, rel_data, rel_url, conn) do
+  def encode_relation({rel_view, rel_data, _rel_url, _conn} = info) do
     %{
-      links: %{
-        self: rel_url,
-        related: rel_view.url_for(rel_data, conn)
-      },
       data: encode_rel_data(rel_view, rel_data)
     }
+    |> merge_related_links(info, Application.get_env(:jsonapi, :remove_links, false))
   end
 
+  defp merge_links(doc, data, view, conn, false = _remove_links) do
+    doc
+    |> Map.merge(%{links: %{self: view.url_for(data, conn)}})
```

Comment:
This doesn't need to be a pipe: https://github.com/christopheradams/elixir_style_guide#avoid-single-pipelines